### ISSUE TITLE
The 2.49.0 version and 2.54.0 are critical incompatible. (silent shutdown..?)

### ISSUE THREAD

**Issue by hmmhmmhm** [ivanhofer/typesafe-i18n#131]
### Describe the bug

Many projects that installed this project lightly as `npm i typesafe-i18n` in version 2.49.0 are currently not working.
So I'm using the version of typesafe-i18n by lowering it to 2.49.0 version.

Can you check this out? I think this is fatal.

### Reproduction

Initialize empty project of polyger (it based typesafe-i18n 2.49.0)
```bash
mkdir some_test
cd some_test
npm init -y
npm install polyger@0.4.14
npx polyger
```
The above command will be nothing executed and silently shutdown. (because some weird silent problem has been occoured in typesafe-i18n.)

then I'll install the previous version of typesafe-i18n and install it.
```bash
npm i typesafe-i18n@2.49.0
npx polyger
```
and this one is working and showing some initialize messages. (if you can confirm it, just you can ctrl+c or command + c to shut it dowm.)

By executing the command below, you can check the previous version as it is.
```bash
npm i typesafe-i18n@2.54.0
npx polyger
```
you can now seeing nothing to executed and can be confirm silent shutdown.

**Comment by ivanhofer** [comment]
Hi, thanks for filing this issue.

When I run `npx polyger` I get following error message:

![image](https://user-images.githubusercontent.com/21335119/143537601-4d081fba-6c76-49c2-b219-15b84a8fe1c2.png)

I'm not sure how you call `i18nObject`. It seems you are passing `undefined` as an argument.

But I don't think that this is the issue that you mean.
Can you please give me more details?

**Comment by hmmhmmhm** [comment]
I tested it in the following environment.

OS: `Windows 10`, `Mac OS X Big Sur`
Node Version: `v12`, `v14`, `v16`

I remember that currently polyger project probably didn't well work in the Linux's OS.
Or you may need to try reproduce after emptying the npm cache using the command below.
```bash
npm cache clean --force
```

<br />


I took the os-locale of and quoted it, but in my case, that part was not a problem.
[locale/index.ts](https://github.com/akasia-dev/polyger/blob/main/locale/index.ts)
```ts
import osLocale from 'os-locale'
import { i18nObject } from './i18n-util'

export const getLocaleCode = async (defaultLocaleCode: Locales) => {
  let localeCode = defaultLocaleCode
  try {
    return (await osLocale()).replace(/-\S+/g, '') as Locales
  } catch (e) {}

  return localeCode as Locales
}

export const getLocale = async (
  localeCode?: Locales,
  options = {
    defaultLocaleCode: 'en' as Locales
  }
) => {
  if (!localeCode) localeCode = await getLocaleCode(options.defaultLocaleCode)
  return i18nObject(localeCode)
}

export default getLocale
```

[src/core/environment.ts](https://github.com/akasia-dev/polyger/blob/main/src/core/environment.ts)
```ts
import getLocale from '../../locale'
const locale = await getLocale()
...
console.log(chalk.magentaBright(locale.polygerIntro()) + '\n\n')
```

If you need it, I will provide you with a virtual environment.

**Comment by ivanhofer** [comment]
I'm running it on `Ubuntu 20.4` and I logged the arguments of the `getLocale` function:

![image](https://user-images.githubusercontent.com/21335119/143541130-bea7cfe6-f944-4dd4-8c1f-5e833f892962.png)

For now I will hardcode the `localeCode` to 'en' to test it.
You probably should use `baseLocale` exported form `./i18n-utils` as a fallback

**Comment by ivanhofer** [comment]
can cou tell me what call inside `polyger` triggers `typesafe-i18n` first?
So in which file and on which line does it fail?

**Comment by hmmhmmhm** [comment]
I'm also trying now, but I failed specify the location itself. It stops like possessed by a ghost.
Instead, this time I'll sharing the situation with VM so that you can test it.

After connecting to each url, press the + button in the center of the right to open the terminal, and enter npm start.

typesafe-i18n@2.49.0 + polyger will works.
https://codesandbox.io/s/polyger-with-typesafe-2-49-vu3ew
<img width="2380" alt="스크린샷 2021-11-26 오후 4 27 05" src="https://user-images.githubusercontent.com/38493722/143543355-15d7eb66-1cf3-4894-b72a-4ffefb1b8108.png">


typesafe-i18n@2.54.0 + polyger will stopped.
https://codesandbox.io/s/polyger-with-typesafe-2-54-hdxwm
<img width="2380" alt="스크린샷 2021-11-26 오후 4 26 37" src="https://user-images.githubusercontent.com/38493722/143543385-254bb0ae-69fa-4256-82f9-716dbb6e4710.png">

**Comment by ivanhofer** [comment]
I have found the issue.
try to replace `createProxy` inside `node_modules/typesafe-i18n/runtime/cjs/runtime/src/util.object.cjs` with this:

```ts
const createProxy =(
	fn,
	prefixKey,
	proxyObject = {},
) =>
	new Proxy((prefixKey ? fn.bind(null, prefixKey) : proxyObject), {
		get: (target, key) =>
			!(target === proxyObject && key === 'then') && createProxy(fn, prefixKey ? `${prefixKey}.${key}` : key),
	})
```
and it sould work again.

**Comment by ivanhofer** [comment]
I thought that this "hack" `(target === proxyObject && key === 'then')` is just angular specific, since I only encountered it there. Something in your code cant handle Proxies correctly. Do you know what this could be? Are you using Proxies inside your code?

**Comment by hmmhmmhm** [comment]
I had confirmed that it was operating normally after modifying the code.
also uploaded it on the in-house CI, and all of them are green.

**Comment by hmmhmmhm** [comment]
+ currently, there is no proxy specified directly in this project.
my colleagues are just guessing that something in the cjs or typescript is twisting it. (zero evidence.)

**Comment by ivanhofer** [comment]
yes, I am also assuming this. If you take a look at the generated code there is some `generator`magic with `switch-case` statements. I need to take a closer look at it in the coming days

**Comment by hmmhmmhm** [comment]
Ok! then, will a temporary fixing version to be release soon?
Thank you for checking quickly so much.
I hope your project will be more successful! 👍🏻

**Comment by ivanhofer** [comment]
@hmmhmmhm thanks for your praise!

Your problem should be fixed in version `2.54.2` that was released a few moments ago
